tests/Repositories/TaskRepositoryTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Tests\Repositories;
+
+use App\Models\Tag;
+use App\Models\Task;
+use App\Models\User;
+use App\Repositories\TaskRepository;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Tests\TestCase;
+
+/**
+ * Class TaskRepositoryTest.
+ */
+class TaskRepositoryTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    /** @var TaskRepository */
+    protected $taskRepo;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->taskRepo = app(TaskRepository::class);
+    }
+
+    /** @test */
+    public function it_can_store_task_with_tags_and_its_assigned_user()
+    {
+        $tag = factory(Tag::class)->create();
+        $assignees = factory(User::class)->times(2)->create();
+        $task = [
+            'title'       => $this->faker->title,
+            'description' => $this->faker->text,
+            'project_id'  => $this->faker->randomDigitNotNull,
```

Comment:
we are writing test cases for this kind of stuff, here some random project id is passed and still test is passing, this should not happen, so put validations that will check given project is exist or not